src/nvim/ex_cmds.c

```diff
@@ -5296,9 +5297,11 @@ void ex_helptags(exarg_T *eap)
   }
 
   /* Get a list of all files in the help directory and in subdirectories. */
-  STRCPY(NameBuff, dirname);
+  if (STRLCPY(NameBuff, dirname, MAXPATHL) >= MAXPATHL)
+    EMSG(_(e_pathtoolong));
```

Comment:
Typically when printing an error, we also return from the function. But here, since we already know `dirname` is a valid directory, we don't need to check for an error.